Modules/rat_facts.py

```diff
@@ -0,0 +1,187 @@
+"""
+rat_facts.py - Managing Facts since 2018
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+
+"""
+from utils.ratlib import Singleton
+from Modules.database_manager import DatabaseManager
+from Modules.context import Context
+import re
+import pyodbc
+import datetime
+
+internal_facts = ("!go", "!assign", "!add")
+# any fact name in here will not be handled by the handler
+
+
+class Fact:
+    name = None
+    language = None
+    message = None
+    author = None
+    timestamp = None
+
+    def __init__(self, name: str, lang: str, message: str, author: str,
+                 timestamp: datetime.datetime):
+        self.name = name
+        self.lang = lang
+        self.message = message
+        self.author = author
+        self.timestamp = timestamp
+
+
+class FactsManager(Singleton):
+
+    @property
+    def enabled(self):
+        return self.dbm.enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        raise RuntimeError("You cant set this. Really. If you want me disable, disable the DBM.")
+
+    def __init__(self):
+        """
+        retrieves the DBM object.
+        """
+        self.dbm = DatabaseManager()
+
+    async def get_fact(self, name: str, lang: str = "en") -> Fact or None:
+        """
+        retrieves a fact from the DB.
+        Args:
+            name: name of the fact to retrieve
+            lang: language to retrieve
+
+        Returns: message of the fact, or None should it not be found for the specified language
+
+        """
+        result = await self.dbm.select_rows("fact", "AND", {"lang": lang, "name": name})
+        timestamp = (await self.dbm.select_rows("fact_timestamps", "AND",
+                                                {"lang": lang, "name": name}))
+        if len(timestamp) > 0:
+            timestamp = timestamp[0].last_modified
+        else:
+            timestamp = datetime.datetime.utcfromtimestamp(0)
+        if result:
+            result = result[0]
+            return Fact(
+                result[0],
+                result[1],
+                result[2],
+                result[3],
+                timestamp
+            )
+        else:
+            return None
+
+    async def is_fact(self, name: str, lang: str = "en") -> bool or None:
+        """
+        checks for existence of a fact
+        Args:
+            name: name of the fact to check
+            lang: language to check
+
+        Returns: True if it exists, False if it exists
+
+        """
+        result = await self.dbm.select_rows("fact", "AND", {"lang": lang, "name": name})
```

Comment:
This introduces duplicity with `get_fact`.
I suggest changing `result` to await `self.get_fact`